### PR TITLE
Theme activation and option details

### LINKED ISSUE #3906: Themes browser: implement the "more options" button for themes
Implement the "more options" button for themes in the theme browser.  The "..." button must open the action bar and not an inline view for iOS

Designs can be found [here]().

### LINKED ISSUE #3907: Allow the user to activate a theme.
Allow the user to activate a theme.

### LINKED ISSUE #3908: Add details web view.
Double check this to see how Android does it.  But @folletto mentioned we should use a web view for the theme details.

### PR DESCRIPTION
**Pull request by alexcurylo**
Closes #3906; closes #3907; closes #3908

Completes theme user experience to match Android implementation

To test: 
1. Enable themes feature with wpdebug://themes?enabled=1 as discussed in #3913
2. Activate themes browser from the blog details screen
3. Check that detail views available for active theme and inactive themes match Android implementation
4. Currently themes may not be purchased through the native apps: check that free and previously purchased themes are available in the browser, and unpurchased themes are not
5. Activate an inactive theme and check that activation is reflected in the browser and on the site
6. Review in general for acceptable congruence with Android themes functionality

@kwonye Please compare to your Android work!
Needs Review: @kwonye

### CODE REVIEW

**Comment by kwonye** [comment]
Wow, this is working great, @alexcurylo. Great job!

Two big picture things are evident:

1. margins/padding of tableviewcells are much different than Android. They should be consistent. @folletto do you mind picking either or?
2. There is no "Activate" rightBarButtonItem when viewing a non-selected theme's details (Try & Customize/View/Details/Support)

As far as the code-level comments, SUPER minor stuff that do not need to be addressed in order to merge (I had to find something, right?)

File: WordPress/Classes/ViewRelated/Themes/ThemeBrowserCell.swift [hunk 1]
```diff
@@ -1,6 +1,53 @@
 import Foundation
 import WordPressShared.WPStyleGuide
 
+/**
+ *  @brief      Actions provided in cell button triggered action sheet
+ */
+public enum ThemeAction {
```
Comment (kwonye) [inline WordPress/Classes/ViewRelated/Themes/ThemeBrowserCell.swift]: Top-level should have `{` on newline

File: WordPress/Classes/ViewRelated/Themes/ThemeBrowserCell.swift [hunk 2]
```diff
@@ -1,6 +1,53 @@
...
+            presenter.presentDetailsForTheme(theme)
+        case .Support:
+            presenter.presentSupportForTheme(theme)
+        case .View:
+            presenter.presentViewForTheme(theme)
+        }
+    }
+}
+
 public class ThemeBrowserCell : UICollectionViewCell {
```
Comment (kwonye) [inline WordPress/Classes/ViewRelated/Themes/ThemeBrowserCell.swift]: Top-level should have `{` on newline

**Comment by folletto** [comment]
> margins/padding of tableviewcells are much different than Android. They should be consistent. @folletto do you mind picking either or?

In general the platform is different thus follows a balance of Calypso standards and iOS standards. The designs are for reference. I can double-check tho...

...but, how do I test this? ;)

**Comment by alexcurylo** [comment]
@kwonye: Design and code feedback addressed, please take another review pass!

Additional tests: select 'Try & Customize' and
7. Note that as per design feedback, no bar 'Activate' is present as one is top left of webview.
8. Activate the theme from inside the webview, and test that the theme browser correctly detects active theme change and updates when dismissed.

**Comment by kwonye** [comment]
Whoo! :shipit:

**Comment by kwonye** [comment]
per @folletto: I noticed just now that if you search a theme, then preview it and go back, it resets the search. Is that avoidable?

**Comment by kwonye** [comment]
Works great @alexcurylo!

In `ThemeBrowserViewController` line 137-154, could you add newlines in-between the functions please?

Once this and the one comment about function parameters is addressed, :shipit:!

**Comment by alexcurylo** [comment]
Thanks for all your review work @kwonye and @folletto!

**Pull request merged**